wagtail/core/rich_text/feature_registry.py

```diff
@@ -96,3 +99,91 @@ def get_converter_rule(self, converter_name, feature_name):
             return self.converter_rules_by_converter[converter_name][feature_name]
         except KeyError:
             return None
+
+
+class HTMLElement(dict):
+    def __init__(self, name: str, is_closing: bool = False, **attrs):
+        self.name = name
+        self.is_closing = is_closing
+        super().__init__(**attrs)
+
+    @property
+    def open_tag(self) -> str:
+        attrs = ''
+        for k, v in sorted(self.items()):
+            if v is not False:
+                attrs += ' %s="%s"' % (escape(k.replace('_', '-')), escape(v))
+        params = (self.name, attrs)
+        if self.is_closing:
+            return '<%s%s />' % params
+        return '<%s%s>' % params
+
+    @property
+    def close_tag(self) -> str:
+        if self.is_closing:
+            return ''
+        return '</%s>' % self.name
+
+
+class LinkHandler:
+    """
+    PageLinkHandler will be invoked whenever we encounter an HTML element
+    in rich text content with an attribute of data-linktype="`linktype`".
+    The resulting element in the database representation will be for example:
+     <a linktype="page" id="42">hello world</a>
+    """
+
+    link_type = None
+    tag_name = 'a'
+
+    @staticmethod
+    def get_model():
+        raise NotImplementedError
+
+    @classmethod
+    def get_instance(cls, attrs: dict) -> Model:
+        model = cls.get_model()
+        try:
+            return model._default_manager.get(id=attrs['id'])
+        except model.DoesNotExist:
+            pass
```

Comment:
@BertrandBordage Any objection to leaving the `DoesNotExist` exception uncaught here, to be handled by the caller? I think that makes it a bit more idiomatic.

Reply:
You mean put it in `find_objects_in_rich_text`? Yeah, that would be valid, I can’t remember a reason why it would be bad.